packages/apps/esm-login-app/src/location-picker/location-picker.component.tsx

```diff
@@ -77,22 +76,28 @@ const LocationPicker: React.FC<LocationPickerProps> = ({ hideWelcomeMessage, cur
 
   const changeLocation = useCallback(
     (locationUuid?: string, saveUserPreference?: boolean) => {
-      setIsSubmitting(true);
+      if (locationUuid) {
+        setIsSubmitting(true);
+      }
 
       const referrer = state?.referrer;
       const returnToUrl = new URLSearchParams(location?.search).get('returnToUrl');
 
-      const sessionDefined = locationUuid ? setSessionLocation(locationUuid, new AbortController()) : Promise.resolve();
+      const sessionDefined = locationUuid ? setSessionLocation(locationUuid, new AbortController()) : Promise.resolve({"redirectToHome": false});
```

Comment:
Instead of all this, it seems like all we need here is:

```suggestion
      const sessionDefined = locationUuid ? setSessionLocation(locationUuid, new AbortController()) : Promise.reject();
```

And then:

```
sessionDefined.then(/* old stuff */).catch(/* not an error, rejected because no location is set */);